api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java

```diff
@@ -102,8 +99,10 @@ public Buildable getBuildable(Request request) throws LegoSetException, ElementN
             EndpointPOJO pojo = ((APIBuildable) buildable).getPojo();
             RequestContext.set(URI, pojo.getUrl());
             RequestContext.set(ENDPOINT_NAME, pojo.getName());
+            RequestContext.set(API_ANNOTATIONS, pojo.getProperties());
             RequestContext.set(ENDPOINT_METHOD, pojo.getHttpMethod());
             ServiceContext.set(ENDPOINT_NAME, pojo.getName());
+            ServiceContext.set(API_ANNOTATIONS, pojo.getProperties());
```

Comment:
Do we really need it in service context? Service context is to be consumed by poseidon framework classes like AbstractServiceClient etc where request context isn't available. We need to be frugal on thread locals.